### PR TITLE
feat: add prompt service

### PR DESCRIPTION
**Pull request by darkskygit**
fix CLOUD-19

### CODE REVIEW

**Comment by darkskygit** [comment]
* **#6528** <a href="https://app.graphite.dev/github/pr/toeverything/AFFiNE/6528?utm_source=stack-comment-icon" target="_blank">Graphite</a>
* **#6527** <a href="https://app.graphite.dev/github/pr/toeverything/AFFiNE/6527?utm_source=stack-comment-icon" target="_blank">Graphite</a>
* **#6506** <a href="https://app.graphite.dev/github/pr/toeverything/AFFiNE/6506?utm_source=stack-comment-icon" target="_blank">Graphite</a>
* **#6525** <a href="https://app.graphite.dev/github/pr/toeverything/AFFiNE/6525?utm_source=stack-comment-icon" target="_blank">Graphite</a>
* **#6445** <a href="https://app.graphite.dev/github/pr/toeverything/AFFiNE/6445?utm_source=stack-comment-icon" target="_blank">Graphite</a>
* **#6448** <a href="https://app.graphite.dev/github/pr/toeverything/AFFiNE/6448?utm_source=stack-comment-icon" target="_blank">Graphite</a>
* **#6437** <a href="https://app.graphite.dev/github/pr/toeverything/AFFiNE/6437?utm_source=stack-comment-icon" target="_blank">Graphite</a>
* **#6272** <a href="https://app.graphite.dev/github/pr/toeverything/AFFiNE/6272?utm_source=stack-comment-icon" target="_blank">Graphite</a>
* **#6230** <a href="https://app.graphite.dev/github/pr/toeverything/AFFiNE/6230?utm_source=stack-comment-icon" target="_blank">Graphite</a>
* **#6254** <a href="https://app.graphite.dev/github/pr/toeverything/AFFiNE/6254?utm_source=stack-comment-icon" target="_blank">Graphite</a>
* **#6241** <a href="https://app.graphite.dev/github/pr/toeverything/AFFiNE/6241?utm_source=stack-comment-icon" target="_blank">Graphite</a> 👈
* **#6229** <a href="https://app.graphite.dev/github/pr/toeverything/AFFiNE/6229?utm_source=stack-comment-icon" target="_blank">Graphite</a>
* `canary`

This stack of pull requests is managed by Graphite. <a href="https://stacking.dev/?utm_source=stack-comment">Learn more about stacking.</a>
<h2></h2>

 Join @darkskygit and the rest of your teammates on <a href="https://graphite.dev?utm-source=stack-comment">Graphite <b>Graphite</b></a>

File: packages/backend/server/schema.prisma [hunk 1]
```diff
@@ -422,6 +422,27 @@ model UserInvoice {
...
+  system
+  assistant
+  user
+}
+
+model AiPrompt {
+  id        String       @id @default(uuid()) @db.VarChar
+  // prompt name
+  name      String       @unique @db.VarChar(20)
+  // if a group of prompts contains multiple sentences, idx specifies the order of each sentence
```
Comment (forehalo) [inline packages/backend/server/schema.prisma:439]: hot to query multiple sentences prompts if their names are unique?

File: packages/backend/server/schema.prisma [hunk 2]
```diff
@@ -422,6 +422,27 @@ model UserInvoice {
...
+enum AiPromptRole {
+  system
+  assistant
+  user
+}
+
+model AiPrompt {
+  id        String       @id @default(uuid()) @db.VarChar
+  // prompt name
+  name      String       @unique @db.VarChar(20)
```
Comment (forehalo) [inline packages/backend/server/schema.prisma]: add a user friendly description column?

File: packages/backend/server/src/plugins/copilot/types.ts [hunk 3]
```diff
@@ -23,8 +23,10 @@ export interface CopilotProvider {
   getCapabilities(): CopilotProviderCapability[];
 }
 
+export const ChatMessageRole = ['system', 'assistant', 'user'] as const;
```
Comment (forehalo) [inline packages/backend/server/src/plugins/copilot/types.ts]: is it possible to directly reference this type from Prisma generated codes?

File: packages/backend/server/schema.prisma (hunk 1 above)
Comment (darkskygit) [reply packages/backend/server/schema.prisma:439]: alreadly fix this in next pr, let me cherry pick to this branch

File: packages/backend/server/schema.prisma (hunk 2 above)
Comment (darkskygit) [reply packages/backend/server/schema.prisma]: if a group of prompts contains multiple prompts, they will be stored as multiple rows in the database.
in this case, it is meaningless to record a description for each row of prompts

currently we only import prompts once through data migration, if we need to manage prompts through the dashboard in the future, we will create a separate table to record these metadata.

**Review by EYHN — APPROVED** [review]

File: packages/backend/server/schema.prisma [hunk 4]
```diff
@@ -422,6 +426,47 @@ model UserInvoice {
...
+  createdAt DateTime     @default(now()) @map("created_at") @db.Timestamptz(6)
+
+  @@unique([name, idx])
+  @@map("ai_prompts")
+}
+
+model AiSession {
+  id          String   @id @default(uuid()) @db.VarChar
+  userId      String   @map("user_id") @db.VarChar
+  workspaceId String   @map("workspace_id") @db.VarChar
```
Comment (forehalo) [inline packages/backend/server/schema.prisma:454]: reference id should be Varchar(36)

Comment (darkskygit) [reply packages/backend/server/schema.prisma:454]: fixed at next pr

**Pull request merged**